src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -41,50 +43,87 @@ const SelectPanelContext = React.createContext<{
   selectionVariant: 'multiple',
 })
 
-// @ts-ignore todo
-const SelectPanel = props => {
-  const anchorRef = useProvidedRefOrCreate(props.anchorRef)
+export type SelectPanelProps = {
+  title: string
+  description?: string
+  selectionVariant?: ActionListProps['selectionVariant'] | 'instant'
+  id?: string
+
+  defaultOpen?: boolean
+  open?: boolean
+  anchorRef?: React.RefObject<HTMLButtonElement>
+
+  onCancel?: () => void
+  onClearSelection?: undefined | (() => void)
+  onSubmit?: (event?: React.FormEvent<HTMLFormElement>) => void
+
+  // TODO: move these to SelectPanel.Overlay or overlayProps
+  width?: AnchoredOverlayProps['width']
+  height?: AnchoredOverlayProps['height']
+
+  children: React.ReactNode
+}
+
+const Panel: React.FC<SelectPanelProps> = ({
+  title,
+  description,
+  selectionVariant = 'multiple',
+  id,
+
+  defaultOpen = false,
+  open: propsOpen,
+  anchorRef: providedAnchorRef,
+
+  onCancel: propsOnCancel,
+  onClearSelection: propsOnClearSelection,
+  onSubmit: propsOnSubmit,
+
+  width = 'medium',
+  height = 'large',
+  ...props
+}) => {
+  const anchorRef = useProvidedRefOrCreate(providedAnchorRef)
 
   // 🚨 Hack for good API!
   // we strip out Anchor from children and pass it to AnchoredOverlay to render
   // with additional props for accessibility
   let renderAnchor: AnchoredOverlayProps['renderAnchor'] = null
   const contents = React.Children.map(props.children, child => {
-    if (child?.type === SelectPanelButton) {
+    if (React.isValidElement(child) && child.type === SelectPanelButton) {
       renderAnchor = anchorProps => React.cloneElement(child, anchorProps)
       return null
     }
     return child
   })
 
-  const [internalOpen, setInternalOpen] = React.useState(props.defaultOpen)
+  const [internalOpen, setInternalOpen] = React.useState(defaultOpen)
   // sync open state
-  React.useEffect(() => setInternalOpen(props.open), [props.open])
+  React.useEffect(() => setInternalOpen(propsOpen || false), [propsOpen])
```

Comment:
It could be worth keeping track of the previous value in state instead of in an effect, I feel like this is similar to the `getDerivedStateFromProps` example from the hooks FAQ: https://legacy.reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops 

Curious if you agree or if there is a difference here 👀

Reply:
Good call! Updated :)